core/src/avm2/globals/string.rs

```diff
@@ -27,13 +40,98 @@ pub fn class_init<'gc>(
     Ok(Value::Undefined)
 }
 
+/// Implements `length` property's getter
+fn length<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Value::String(s) = this.value_of(activation.context.gc_context)? {
+            return Ok(s.encode_utf16().count().into());
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `String.charAt`
+fn char_at<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    match char_code_at(activation, this, args)? {
+        Value::Number(code) if code.is_finite() => Ok(AvmString::new(
+            activation.context.gc_context,
+            String::from_utf16_lossy(&[code as u16]),
+        )
+        .into()),
+        Value::Number(_) => Ok("".into()),
+        _ => Ok(Value::Undefined),
+    }
+}
+
+/// Implements `String.charCodeAt`
+fn char_code_at<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Value::String(s) = this.value_of(activation.context.gc_context)? {
+            let mut index = args
+                .get(0)
+                .unwrap_or(&Value::Number(0.0))
+                .coerce_to_number(activation)?;
+
+            if index < 0.0 {
+                return Ok(std::f64::NAN.into());
+            }
+            if index.is_infinite() {
+                index = 0.0;
+            }
+
+            let code: f64 = s
+                .encode_utf16()
+                .collect::<Vec<u16>>()
+                .get(index as usize)
+                .cloned()
+                .map(|v| v as f64)
+                .unwrap_or(std::f64::NAN);
+
+            return Ok(code.into());
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
 /// Construct `String`'s class.
 pub fn create_class<'gc>(mc: MutationContext<'gc, '_>) -> GcCell<'gc, Class<'gc>> {
-    Class::new(
+    let class = Class::new(
         QName::new(Namespace::public(), "String"),
         Some(QName::new(Namespace::public(), "Object").into()),
         Method::from_builtin(instance_init),
         Method::from_builtin(class_init),
         mc,
-    )
+    );
+
+    let mut write = class.write(mc);
+
```

Comment:
String should be final and non-dynamic:
`write.set_attributes(ClassAttributes::FINAL | ClassAttributes::SEALED);`
(even though we may not honor these yet).